community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseFactoryState.java

```diff
@@ -75,6 +81,11 @@ public void addKernelExtensions( Iterable<KernelExtensionFactory<?>> newKernelEx
         }
     }
 
+    public void addURLAccessRule( String protocol, URLAccessRule rule )
```

Comment:
I also wasn't sure where to put `URLAccessRule`, however I chose under `org.neo4j.kernel` because the other dependencies that are injected into `GraphDatabaseFactoryState` are also there (`org.neo4j.kernel.extension.KernelExtensionFactory` and `org.neo4j.kernel.monitoring.Monitors`). I also figured it was more about the kernel than about the graph, so that swayed me.

Not sure what you mean about this being possible via `GraphDatabaseSettings`. AFAIK, there is no way to inject an concrete object that way, nor is there any setting that influences the allowed protocols - which is needed for the acceptance test here: https://github.com/cleishm/neo4j/blob/2.3-url-access-rules/community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/LoadCsvAcceptanceTest.scala#L321-L335